packages/core/src/StateNode.ts

```diff
@@ -1151,9 +1156,9 @@ class StateNode<
 
   private resolveTransition(
     stateTransition: StateTransition<TContext, TEvent>,
-    currentState?: State<TContext, TEvent, any, any>,
-    _event: SCXML.Event<TEvent> = initEvent as SCXML.Event<TEvent>,
-    context: TContext = this.machine.context
+    currentState: State<TContext, TEvent, any, any> | undefined,
+    context: TContext,
+    _event: SCXML.Event<TEvent> = initEvent as SCXML.Event<TEvent>
```

Comment:
arguments were reordered to make `context` **required** and a required param can't follow an optional one (`_event`)

Reply:
oh, and this was OK to do because `resolveTransition` is a private method, so it's not a breaking change